.github/workflows/example-scan-image.yaml

```diff
@@ -0,0 +1,29 @@
+name: Kubescape scanning for image vulnerabilities
+on: [push, pull_request]
+jobs:
+  kubescape-scan-image:
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+    steps:
+    - uses: actions/checkout@v3
+    - uses: kubescape/github-action@main
+      continue-on-error: true
+      with:
+        image: quay.io/kubescape/kubescape
+        format: sarif
+        outputFile: results.sarif
+        # severityThreshold: "critical"
+        # # Username for a private registry with the image
+        # registryUsername: ${{secrets.KUBESCAPE_REGISTRY_USERNAME}}
```

Comment:
```suggestion
        # registryUsername: ${{secrets.REGISTRY_USERNAME}}
```

Reply:
I don’t think we should change this because it blurs the intent of the secret. Is it a generic registry username or the account for the image scanner or my CI service?

If I use a repo to build a web application and I push the image there, good practice is to have tools separated by accounts, so there is a differentiated trail of who accesses what and when. It also promotes clarity to users what’s there.

If people feel this is excessive, they can rename the variables as they see fit. But if there’s a good practice to follow and it doesn’t twist your arms, we should nudge users towards it.

Reply:
On the other hand, it is confusing the `KUBESCAPE_REGISTRY_USERNAME`. At first glance I didn't understand why a user needs the credentials of the Kubescape registry.

We could change it to `secrets.SCAN_REGISTRY_USERNAME` or something similar (if you have a better idea)

Reply:
You have a point. I like your idea. I think it would work best if we update the matching comment to explain the meaning better and touch up the grammar a bit:

```suggestion
        # # Username for the image registry account you use to retrieve and scan images
        # registryUsername: ${{secrets.SCANNER_REGISTRY_USERNAME}}
```

How’s that sound?

Reply:
Or `you use to retrieve images for scanning`.

Reply:
I think that writing `Scanner` is very similar to writing `Kubescape`, because it refers to the tool/scanner while in reality this secret is related to the scanned image, and not the scanner tool.

Reply:
> while in reality this secret is related to the scanned image, and not the scanner tool

Why does it have to relate to only one of them?

In my view, it’s not a generic one-and-only username for _Kubescape’s_ registry. It’s _Kubescape’s_ account for your registry. You can also have Grype’s account for the same registry. Because registry username and password identify _who_ performs actions, not _what_ they work on.

Let’s say you use Kubescape and Grype to scan images in CI. One day someone attacks your registry with stolen credentials. You go to the action log and you’d like to see info on the leaked credentials. If you use an account per tool, you get a clear picture of where the leak is connected to. If you use an account per registry, you see no useful information to aid your investigation.

Which is why it’s helpful to have separate credentials for separate tools. And the same reason you and I have different accounts (credentials, username and password) for our Github accounts: to see _who_ does what, not that someone with access to our hypothetical one-for-all Github account deleted every repo in the organization.

But whatever, it’s not a hill I die on. Nothing like variable naming to start a debate, huh? :sweat_smile:

Reply:
You are mentioning here best practice, but the idea of the readme is to explain to the user what he needs to do.
Now, since I was confused when I saw the readme, and I thought that the secret refers to kubescape credentials, that means that the readme is unclear. 
We should focus on making the readme clear and not educating users with best practices.